File: src/types/report.ts
import type { CleanupResultRecord, ScanSummary } from "./package";

export type ReportSummary = {
  deviceLabel: string;
  exportStatus: string;
  id: string;
  summary: string;
  timestamp: string;
};

export type CleanupSessionReport = {
  afterSummary: ScanSummary;
  beforeSummary: ScanSummary;
  deviceLabel: string;
  exportStatus: string;
  id: string;
  launcherObservations: string[];
  results: CleanupResultRecord[];
  selectedPackages: string[];
  summary: string;
  timestamp: string;
};

export type PdfExportLanguage = "en" | "no";

export type PdfExportResult = {
  exportStatus: string;
  message: string;
  path: string | null;
  previewDataUrl: string | null;
  success: boolean;
};

File: src-tauri/src/models/report.rs
use serde::{Deserialize, Serialize};

use super::package::ScanSummary;

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ReportSummary {
    pub id: String,
    pub device_label: String,
    pub summary: String,
    pub timestamp: String,
    pub export_status: String,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CleanupSessionReport {
    pub after_summary: ScanSummary,
    pub before_summary: ScanSummary,
    pub id: String,
    pub device_label: String,
    pub launcher_observations: Vec<String>,
    pub selected_packages: Vec<String>,
    pub results: Vec<crate::models::package::CleanupResultRecord>,
    pub summary: String,
    pub timestamp: String,
    pub export_status: String,
}

#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
#[serde(rename_all = "lowercase")]
pub enum PdfExportLanguage {
    En,
    No,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PdfExportResult {
    pub success: bool,
    pub path: Option<String>,
    pub message: String,
    pub export_status: String,
    pub preview_data_url: Option<String>,
}
